engine/step/step.go

```diff
@@ -264,6 +300,59 @@ func Run(st *Step, baseConfig map[string]json.RawMessage, values *values.Values,
 		case <-stopRunningSteps:
 			st.State = StateToRetry
 		default:
+			for _, hookName := range st.PreHooks {
+				h, err := getHook(hookName)
+				if err != nil {
+					st.Error = fmt.Sprintf("get hook %s failed: %v", h.Name, err)
+					st.State = StateFatalError
+					go noopStep(st, stepChan)
+					return
+				}
+
+				o, m, err := runner.Exec(h.Name, baseCfgRaw, h.Action.Configuration, ctx)
```

Comment:
where does this runner is coming from ? I'm afraid, it comes from `runner, err := getRunner(st.Action.Type)` so it wouldn't be the runner needed for the hooks' actions.